troi/patches/new_jams.py

```diff
@@ -0,0 +1,80 @@
+from datetime import datetime
+
+import troi.filters
+import troi.listenbrainz.feedback
+import troi.listenbrainz.listens
+import troi.listenbrainz.recs
+import troi.musicbrainz.recording_lookup
+from troi import Playlist, Element, Recording
+from troi.musicbrainz.recording import RecordingListElement
+from troi.playlist import PlaylistMakerElement, PlaylistShuffleElement
+
+DAYS_OF_RECENT_LISTENS_TO_EXCLUDE = 60  # Exclude tracks listened in last X days from the daily jams playlist
+DAILY_JAMS_MIN_RECORDINGS = 25  # the minimum number of recordings we aspire to have in a daily jam, this is not a hard limit
+BATCH_SIZE_RECS = 100  # the number of recommendations fetched in 1 go
+MAX_RECS_LIMIT = 1000  # the maximum of recommendations available in LB
+
+
+class NewJamsPatch(troi.patch.Patch):
+    """
+        Generate a playlist from the ListenBrainz recommended recordings that the user has not listened to.
+    """
+
+    def __init__(self, debug=False):
+        super().__init__(debug)
+
+    @staticmethod
+    def inputs():
+        """
+        Generate a playlist from the ListenBrainz recommended recordings that the user has not listened to.
+
+        \b
+        USER_NAME is a MusicBrainz user name that has an account on ListenBrainz.
+        """
+        return [
+            {"type": "argument", "args": ["user_name"]}
+        ]
+
+    @staticmethod
+    def outputs():
+        return [Playlist]
+
+    @staticmethod
+    def slug():
+        return "new-jams"
+
+    @staticmethod
+    def description():
+        return "Generate a playlist from the ListenBrainz recommended recordings that the user has not listened to."
+
+    def create(self, inputs):
+        user_name = inputs['user_name']
+
+        recs = troi.listenbrainz.recs.UserRecordingRecommendationsElement(user_name, "raw", count=1000)
```

Comment:
Can you test whether pylistenbrainz automatically converts this into batch of calls because IIRC LB API will return 100 at max in one call so either pylb or a loop here has to batch calls accordingly.

Reply:
1000 tracks are actually returned, yes. But I will triple check before merging.

Reply:
It isn't pylistenbrainz that does, but datasethoster that does:

https://github.com/metabrainz/troi-recommendation-playground/blob/main/troi/listenbrainz/recs.py#L37

And I've been listening to the new daily jams for 5 days now, solid, with much better results.